### PR TITLE
Implement my own Array#my_map, Array#my_select, Array#my_each

### PR DESCRIPTION
**Pull request by andreyruby**

### CODE REVIEW

File: HW02/andrey_bakanovsky/hw02.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module Enumerable_Refine # defined the module which class we are refining, whith an new name
+    refine Array do # the Class is refining
+        # implement a new own method
```
Comment (B0RlS) [inline HW02/andrey_bakanovsky/hw02.rb:3]: Please remove all comments from the code.
If you like, you can add the readme.md file to your PR

File: HW02/andrey_bakanovsky/hw02.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+module Enumerable_Refine # defined the module which class we are refining, whith an new name
```
Comment (B0RlS) [inline HW02/andrey_bakanovsky/hw02.rb:1]: Could you run `rubocop -a ` and fix styles errors please?

File: HW02/andrey_bakanovsky/hw02.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        # the method creates a new array which contains elements
+        #  that conforms the block conditions
+        def my_select
+            new_array = []
+            if block_given?
+                i = 0
+                while i < self.length 
+                    if yield self[i]
+                        new_array << self[i]
+                    end
```
Comment (B0RlS) [inline HW02/andrey_bakanovsky/hw02.rb:26]:
```suggestion
                  new_array << self[i] if yield self[i]
```

**Review by vladhilko — APPROVED** [review]
Could you run `rubocop -a` ?

File: HW02/andrey_bakanovsky/hw02.rb (hunk 2 above)
Comment (B0RlS) [inline HW02/andrey_bakanovsky/hw02.rb:1]: Make module/class names in Camelcase style please

Comment (B0RlS) [inline HW02/andrey_bakanovsky/hw02.rb:1]: Your module name must be the same as the filename

Comment (B0RlS) [inline HW02/andrey_bakanovsky/hw02.rb:1]: Could you explain to me the name of your module, why did you choose this name?

File: HW02/andrey_bakanovsky/hw02.rb (hunk 1 above)
Comment (andreyruby) [reply HW02/andrey_bakanovsky/hw02.rb]: Ok, I fixed it.

File: HW02/andrey_bakanovsky/hw02.rb (hunk 2 above)
Comment (andreyruby) [reply HW02/andrey_bakanovsky/hw02.rb]: Ok, I ran it and fixed errors.

File: HW02/andrey_bakanovsky/hw02.rb (hunk 3 above)
Comment (andreyruby) [reply HW02/andrey_bakanovsky/hw02.rb]: I agree, thanks.

File: HW02/andrey_bakanovsky/hw02.rb (hunk 2 above)
Comment (andreyruby) [reply HW02/andrey_bakanovsky/hw02.rb]: Thanks, I fixed it.

Comment (andreyruby) [reply HW02/andrey_bakanovsky/hw02.rb]: Just I tried to implement a new module similar to module Enumerable - Enumerable_Refine with own methods for class Array. Now I suppose that it's possible incorrect and I changed module name and filename.

File: HW02/andrey_bakanovsky/main.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require_relative 'hw02'
+using Enumerable_Refine
```
Comment (vladhilko) [inline HW02/andrey_bakanovsky/main.rb:4]: You forgot to update name here

Comment (andreyruby) [reply HW02/andrey_bakanovsky/main.rb]: Ok, I updated it. Thanks.

File: HW02/andrey_bakanovsky/hw02.rb (hunk 2 above)
Comment (andreyruby) [reply HW02/andrey_bakanovsky/hw02.rb]: I fixed it.

**Pull request merged**